### DISCUSSION TITLE
Ability to specify files lifetime per attribute filter

### DISCUSSION THREAD

**Discussion by builder-main** [git-lfs/git-lfs#5782] [Ideas]
We'd like a way to specify GIT LFS to keep only a certain number of versions. Could this be done by recycling the pointers for instance ?

Motivation :
- Better the workflow for game devs and avoid bloating their repositories
- Avoid having to use BFS or git filter repo on a regular basis.
- Keep git and git lfs as a single versioning tool.
- Use git lfs with a diversity of files, sources and compiled, with differente requirements, (Think PSD vs JPG for instance, or Blender files vs FBX)
- No history rewrite (as otherwise it would be server behaviour dependant,  and we can see that gitlab / bitbucket / github don't all have the same policy)
 
The discussion is open.

**Comment by chrisd8088** [comment]
Hey, thanks for the suggestion!  We're always interested in proposals for new functionality.  If anyone has time to write up a detailed proposal in a PR, we can add it to the `docs/proposals` [section](https://github.com/git-lfs/git-lfs/tree/main/docs/proposals) of our documentation.

I will say that we would almost certainly want to avoid any functionality that would remove content automatically and without user confirmation, since is likely going to result in accidental loss of data.

However, if the options in the `git-lfs-prune(1)` [command](https://github.com/git-lfs/git-lfs/blob/41612f7cb5de921303534142ee5bc51444df670f/docs/man/git-lfs-prune.adoc) don't provide a sufficient range for your use case, perhaps we could add support for automatic pruning of the local cache based on an additional Git attribute?

That said, there are some known bugs in our handling of Git macro attributes (e.g., #5179, #5181, #5383), which it would be nice to fix before we move on to implementing other attribute-based logic.
